tests/fsharp/optimize/tests_optimize.fs

```diff
@@ -0,0 +1,276 @@
+module ``FSharp-Tests-Optimize``
+
+open System
+open System.IO
+open NUnit.Framework
+
+open NUnitConf
+open PlatformHelpers
+open FSharpTestSuiteTypes
+
+let testContext = FSharpTestSuite.testContext
+
+module Analyses = 
+
+    let ``fsc >a 2>&1`` cfg dir = 
+        let ``exec >a 2>&1`` outFile p = 
+            Command.exec dir cfg.EnvironmentVariables { Output = OutputAndError(Overwrite(outFile)); Input = None; } p 
+            >> checkResult
+        Printf.ksprintf (fun flags sources out -> Commands.fsc (``exec >a 2>&1`` out) cfg.FSC flags sources)
+
+    let fsdiff cfg dir a b = processor {
+        let out = new ResizeArray<string>()
+        let redirectOutputToFile path args =
+            log "%s %s" path args
+            let toLog = redirectToLog ()
+            Process.exec { RedirectOutput = Some (function null -> () | s -> out.Add(s)); RedirectError = Some toLog.Post; RedirectInput = None; } dir cfg.EnvironmentVariables path args
+        do! (Commands.fsdiff redirectOutputToFile cfg.FSDIFF true a b) |> (fun _ -> Success ())
+        return out.ToArray() |> List.ofArray
+        }
+
+    [<Test; FSharpSuiteTest("optimize/analyses")>]
+    let functionSizes () = check (processor {
+        let { Directory = dir; Config = cfg } = testContext ()
+
+        let getfullpath = Commands.getfullpath dir
+        let ``fsc >a 2>&1`` = ``fsc >a 2>&1`` cfg dir  
```

Comment:
We generally don't use this "partially apply a function and bind it to the same name" pattern in the F# compiler codebase.  Why doesn't testContext return an object that has these various helpers as members or extension members?  then the whole code becomes

    let functionSizes() = check (processor {
        let ctxt = textContext()
        ...
        do! ctxt.``fsc >a 2>&1`` ....
        ...
    }

That's just more succinct and less code to maintain.